### DISCUSSION TITLE
About language translation

### DISCUSSION THREAD

**Discussion by cneve** [fra589/cn5X#28] [General]
Hello, recently, I have encountered problems in language translation.
I copied a cn5X.en.ts file, and renamed it, and translated the characters using Linguist, and finally published the QM file using Linguist,which was saved in the same directory as the TS file.
The corresponding language selection is already shown on the software menu, but when I click the button, the command window appears "Locale () not usable, using default to english"
If I want to make a local language,What should I do next?

**Comment by fra589** [comment]
Hi @cneve 
You also need to update the cn5X_locales.xml file in the i18n directory...
In wich language do you have made your translation ? It will be fine if you can send it to me :-)
@++;
Gauthier.

**Comment by cneve** [comment]
I am now changing in the en.ts file. After I have finished, I will send you the TS file. Gauthier thank you for your help.

**Comment by luisluna2** [comment]
@fra589 
Friend, how can traslate?, with what program?

**Comment by fra589** [comment]
> how can traslate?, with what program?

Hi @luisluna2,
You don't need to make any change in the Python .py files.
- First, install the QT linguist software,  
- Then, go to the \<cn5X\>/i18n directory,  
- Duplicate the file cn5X.en.ts to cn5X.es.ts for example to make the spanish translation (the actual cn5X.es.ts is an empty translated file, so you can replace it),  
- Open your new .ts file with QT linguist software,  
- Change the language cible in the "Edit/File parameter" menu,  
- Replace all the items english translation by your new langage translation,  
- Save the new translated .ts file,
- Use the "File/Publish" menu to publish the .qm file in the same \<cn5X\>/i18n directory. It is this cn5X.\<LOCALE\>.qm wich is used by cn5X to translate the interface at run time,  
- Don't forget to update the i18/cn5X_locales.xml to reflect your new translation,  
- Dont forget to send to me your new .ts file :-)  

Thanks for your contribution,

@++;
Gauthier.

**Comment by fra589** [comment]
Hi all,

To avoid duplicate work, please, write here the translation you plan to do.

By the way, don't modify the files : mainWindow.py, dlgConfig.py and dlgAPropos.py. Those files are generated by the QT Designer devloppement tool and probably will be crushed by the next ui change.

I make the choice of don't translate some of the ui items, especially the originals Grbl commands or messages wich stay in the Grbl language (for example the "Soft Reset (^X)" button). This can be discuss here :-)


Thanks,
Gauthier.

**Comment by cneve** [comment]
Gauthier ，if  translate some of the original code of grbl, will there be problems?Grblcom connection failure occurs several times when clicking connect.
Today, I translated most of the buttons and labels，I found many failed connections, which led me to reinstall grbl.˃̣̣̥᷄⌓˂̣̣̥᷅

**Comment by fra589** [comment]
HI @cneve,  
  
If you translate the grbl code itself, cn5x to grbl serial com will stop to work since cn5X analyse and wait for the original sentences of grbl.  
For example, to complete the connection, cn5X wait to receive a string (case sensitive) wich begin with "Grbl " and end with "help]"... So, if you translate the word "help" in the original code of grbl, cn5X never end its connection.
  
however, when you made many connections and deconnections, grbl crash ans stop to work correctly (I don't know why but it's a fact). In this case, generaly, it's enought to power off the grbl card and close cn5X, wait few seconds and restart the operation.  
  
@++;  
Gauthier.

**Comment by luisluna2** [comment]
Hi @fra589 I will modify to Spanish :)

**Comment by dbxzjq** [comment]
hi @fra589 
 CN5X can publish WINX86 installation file, thank you!

**Comment by fra589** [comment]
Hi @dbxzjq, 
I am sorry, but my Windows develop box is broken (pleonasm :yum:), so, I can't make Windows installation files for the moment(and probably for a long time).
But you can install prerequisites separately and run cn5X directly after...  
  
To install on Windows :
- Download and install latest Python 3 from https://www.python.org (don't forget to add Python in PATH environment variable)
- Run cmd (as Administrator if you have installed Python for all users)
- in cmd, type "pip install pyqt5"
- Download and copy the whole cn5X directory from Github,  
  
=> Double click on the cn5X.py downloaded file in Windows explorer... 

Enjoy using cn5X++ 

@++;
Gauthier.

**Comment by Andarlo** [comment]
bonjour  petite présentation :
je suis tourneur de profession depuis plus de 10 ans et je me lance dans la fabrication de ma 2nd 
cnc fraiseuse . la 1er étant faite de règle de maçon et impression 3d ( pas assez solide l'usinage de petite 
 pièce d'alu) .
Pour celle ci en rexrot et vis de Ø18 et rail de 20mm sa devrait le faire , pour une dimension environ 1700x1500 .
Pour la partit électronique j'ai  2 méga 2560 ( une en 4x et l'autre en 6x )
je compte m'installer un petit mandrin sur axe a et b pour le plaisir .
donc démarrage par 4x puis upgrade en 6x.

avez vous de nouvelle avancer pour grbl 5x sous Windows ?
quelle serait les risque de l'utilise a l'heure actuelle en usinage simple 3 axes ?
merci pour ce superbe travail.

**Comment by fra589** [comment]
Bonjour @Andarlo,  
  
Il aurait préférable d'ouvrir ton propre sujet. Plutôt comme [discussion](https://github.com/fra589/cn5X/discussions) que comme [issue](https://github.com/fra589/cn5X/issues), vu qu'il ne s’agit pas à proprement parler de bug ou problème d'utilisation.  
  
Depuis que j'ai réécrit le module de communication en utilisant le module pyserial à la place de celui de QT (QSerial), cn5X++ fonctionne sous Windows à priori aussi bien que sous Linux (d'après les retours que je peux avoir).  
  
Je l'utilise personnellement pour l'usinage réel de toutes mes pièces fraisées depuis maintenant plusieurs années. Il reste probablement quelques bugs, mais je ne suis pas encore tombés dessus. Alors [merci de les faire remonter](https://github.com/fra589/cn5X/issues) si tu en découvre de nouveaux 😃   
  
@++;  
Gauthier.

**Reply by Andarlo** [reply]
bonjour ok dsl pour le post pouvez vous peut être le déplacer au besoin .
avez vous des skin pour l'interface car j'ai installer cn5x pour tester et je ne retrouve pas le même thème que sur les diffèrent post de discutions ?
cordialement

**Comment by fra589** [comment]
Bonjour @Andarlo,  
  
Je n'ai pas prévu pour l'instant de personnalisation d’apparence tel que les "skins".
Tu as peut être vu des copies d'écran de versions précédentes...
L'interface graphique évolue au fur et à mesure de l'ajout des différentes fonctionnalités, de plus, vue que je développe sous Linux, mes copies d'écran n'ont peut être pas exactement le même look que quand le programme tourne avec Windows.
Voici a quoi ressemblera la prochaine version (ajout des boutons G28 & G30 et des affichage de status d'entrées digitales supplémentaires):  
  
![image](https://user-images.githubusercontent.com/19529154/155084989-c3de420f-932c-4526-bc03-7f1a8bc0e26f.png)
  
![image](https://user-images.githubusercontent.com/19529154/155084731-0619f2f5-7874-4d26-adc1-0c8dcaa0a7d2.png)  
  
@++;  
Gauthier.

**Reply by Andarlo** [reply]
je te fait une impression écran des que je peux car sous Windows je n'est vraiment pas tout c'est bouton .
je crois avoir lu que l'on pouvait installer la version linux sous Windows en installant python 3.

**Comment by fra589** [comment]
https://github.com/fra589/cn5X/wiki/Install-cn5X-on-Windows
Je n'ai pas encore publié la dernière version avec G28/G30 et les entrées digitales.

**Comment by Andarlo** [comment]
![Sans titre](https://user-images.githubusercontent.com/99893235/155145397-0ba09ccd-8915-478b-afc6-8e7fb1b3e398.png)

**Comment by Andarlo** [comment]
voici a quoi ressemble mon interface

**Comment by fra589** [comment]
Ben, c'est normal que ça ne ressemble pas à cn5X, puisque ce n'est pas le même logiciel.  
  
La copie d'écran que tu as postée correspond à [GRBL HotWire Mega 5X](https://rckeith.co.uk/grbl-hotwire-mega-5x-for-cnc-foam-cutters/), un autre logiciel qui pilote le firmware grbl-Mega-5X pour faire de la découpe au fil chaud.  
  
cn5X++ a été écrit pour faire du fraisage CNC jusqu'à 6 axes avec grbl-Mega-5X.  
  
@++;  
Gauthier.

**Comment by Andarlo** [comment]
ok je me suis tromper de fichier alors le lien source pour Windows ce trouve sur git?

**Comment by fra589** [comment]
?!?!?!
Ben... T'es dessus : https://github.com/fra589/cn5X

Pour installer : https://github.com/fra589/cn5X/wiki/Install-cn5X-on-Windows

**Reply by Andarlo** [reply]
il y a une méthode plus simple que l'invite cmd pour faire l'Install?

**Comment by Andarlo** [comment]
qu'est ce que Ne pas oublier d'ajouter Python au PATH
